src/plrust.rs

```diff
@@ -344,19 +367,23 @@ fn plrust_fn_{fn_oid}"#
     source.push_str(" {\n");
     source.push_str(&code);
     source.push_str("\n}");
-    source
+    Ok(source)
 }
 
+#[tracing::instrument(level = "debug")]
 fn extract_code_and_args(
     fn_oid: pg_sys::Oid,
-) -> (
-    pg_sys::Oid,
-    String,
-    String,
-    Vec<(PgOid, Option<String>)>,
-    (PgOid, bool),
-    bool,
-) {
+) -> Result<
+    (
+        pg_sys::Oid,
+        String,
+        String,
+        Vec<(PgOid, Option<String>)>,
+        (PgOid, bool),
+        bool,
+    ),
+    PlRustError,
+> {
```

Comment:
I begin to resent this large return tuple. Its girth mocks me. It was already like that, however.

Reply:
That's my next task after #46, I was hoping to give you a little idea we could find consensus over first though!